.github/workflows/push_ci_docker_base.yml

```diff
@@ -40,3 +40,23 @@ jobs:
           path: clang-tidy
           aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
           aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+  cilint-clang-tidy-macos:
+    runs-on: macos-10.15
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Build binary
+        run: |
+          cd clang-tidy-checks
+          ./setup.sh
+      - uses: driazati/upload-artifact-s3@50adbe4ef0b6d9221df25c18c5fc528dfcb7c3f8
+        if: github.event_name == 'push' && github.ref == 'refs/heads/master'
```

Comment:
Wait I noticed you already have the guarding set up. Is there something bad about enabling this flow on PRs as well?

Reply:
Github doesn't allow you to access credentials on PRs for security reasons

Reply:
I don't follow. Wouldn't this step not happen if the event trigger isn't push? Or are you saying another step requires the credentials?

Reply:
yeah according to the top of this file, this workflow only runs on `push`, right?

Reply:
we could just have this run on `pull_request` too then it would do the builds but not push (which sounds like what we want)

Reply:
Yes that was what I intended to suggest